Intersect.Editor/Core/Graphics.cs

```diff
@@ -1388,65 +1388,122 @@ bool alternate
                         continue;
                     }
 
-                    if (TextUtils.IsNone(resource.Initial.Graphic))
+                    if (resource.HealthGraphics.Count == 0)
                     {
                         continue;
                     }
 
-                    if (resource.Initial.GraphicFromTileset)
+                    var fullVitalResourceGraphic = resource.HealthGraphics.FirstOrDefault(
+                        g => g.Value.MaxHp == 100
+                    );
+
+                    if (fullVitalResourceGraphic.Value is not { MaxHp: 100 } resourceGraphic)
```

Comment:
Please don't use KeyValuePairs when they don't make sense, you should be doing `.Values` before `FirstOrDefault()`